### PR TITLE
major update(functions & UI)

### PR DESCRIPTION
**Pull request by INSIinc**
1. add Nextcloud text editor integration
2. add attachment support
3. support attachment viewer & download
4. classify announcements by time
5. remake user interface(create form,announcement list and more)
6. add real everyone group at beginning

AnnouncementCenterFrontpage

### CODE REVIEW

**Comment by nickvergessen** [comment]
Oh wow, that's some fancy and big update!
I invited you to our organisation, then you can push the branch directly to the repo and CI will run happily all the time and at the same time we can collaborate on some last fixes.

> 1. add Nextcloud text editor integration

That's very fancy, nice!

> 2. add attachment support
> 3. support attachment viewer & download

I don't really need that for my usecases, but it's also cool to have!

> 4. classify announcements by time

I guess that's the left sidebar. I was always dreaming of this, but never found the time to implement it. 🤩  Maybe we can migrate that to `[NcAppNavigationCaption](https://nextcloud-vue-components.netlify.app/#/Components/App%20containers/NcAppNavigation?id=ncappnavigationcaption)` for the headings and [NcListItem](https://nextcloud-vue-components.netlify.app/#/Components/NcListItems?id=nclistitem) for the announcements? :)

Nextcloud Designers also offer Design feedback and app reviews. Every Tuesday 13 UTC. Not sure that is too late for you (guessing your time zone from the language of the screenshots). If you are up for it, I would ask them to give us a spot and then we can walk it through with them at https://cloud.nextcloud.com/index.php/call/gqff69i8

---

There are some technical difficulties with your PR (chmod of all files changed etc) which I guess is because you commited from Windows? It's not a problem, we can revert this together :)

**Comment by INSIinc** [comment]
I am sorry that I forgot to deal with this update because I was busy with a voice translation service. I am really sorry that I have come back now...

**Comment by INSIinc** [comment]
> Oh wow, that's some fancy and big update!哦，哇，这是一些花式和大更新！ I invited you to our organisation, then you can push the branch directly to the repo and CI will run happily all the time and at the same time we can collaborate on some last fixes. 我邀请您加入我们的组织，然后您可以将分支直接推到存储库，CI将一直愉快地运行，同时我们可以在一些最后的修复上进行协作。
> 
> > 1. add Nextcloud text editor integration添加Nextcloud文本编辑器集成
> 
> That's very fancy, nice!很漂亮，很好！
> 
> > 2. add attachment support 添加附件支持
> > 3. support attachment viewer & download支持附件查看器和下载
> 
> I don't really need that for my usecases, but it's also cool to have!我真的不需要我的用例，但它也很酷！
> 
> > 4. classify announcements by time按时间分类公告
> 
> I guess that's the left sidebar. I was always dreaming of this, but never found the time to implement it. 🤩 Maybe we can migrate that to [NcAppNavigationCaption](https://nextcloud-vue-components.netlify.app/#/Components/App%20containers/NcAppNavigation?id=ncappnavigationcaption) for the headings and [NcListItem](https://nextcloud-vue-components.netlify.app/#/Components/NcListItems?id=nclistitem) for the announcements? :)我想这是左边栏。我一直梦想着这一点，但从来没有找到时间来实现它。也许我们可以迁移到NcAppNavigationCaption的标题和NcListItem的公告？：）
> 
> Nextcloud Designers also offer Design feedback and app reviews. Every Tuesday 13 UTC. Not sure that is too late for you (guessing your time zone from the language of the screenshots). Nextcloud设计师还提供设计反馈和应用程序评论。每周二13 UTC。不知道这对你来说是否太晚了（从截图的语言中猜测你的时区）。 If you are up for it, I would ask them to give us a spot and then we can walk it through with them at https://cloud.nextcloud.com/index.php/call/gqff69i8如果你准备好了，我会请他们给予我们一个位置，然后我们可以在https://cloud.nextcloud.com/index.php/call/gqff69i8上与他们一起走过。
> 
> There are some technical difficulties with your PR (chmod of all files changed etc) which I guess is because you commited from Windows? It's not a problem, we can revert this together :)有一些技术上的困难与您的公关（chmod的所有文件更改等），我猜这是因为你从Windows提交？这不是一个问题，我们可以一起恢复：）



> Oh wow, that's some fancy and big update!哦，哇，这是一些花式和大更新！ I invited you to our organisation, then you can push the branch directly to the repo and CI will run happily all the time and at the same time we can collaborate on some last fixes. 我邀请您加入我们的组织，然后您可以将分支直接推到存储库，CI将一直愉快地运行，同时我们可以在一些最后的修复上进行协作。
> 
> > 1. add Nextcloud text editor integration添加Nextcloud文本编辑器集成
> 
> That's very fancy, nice!很漂亮，很好！
> 
> > 2. add attachment support 添加附件支持
> > 3. support attachment viewer & download支持附件查看器和下载
> 
> I don't really need that for my usecases, but it's also cool to have!我真的不需要我的用例，但它也很酷！
> 
> > 4. classify announcements by time按时间分类公告
> 
> I guess that's the left sidebar. I was always dreaming of this, but never found the time to implement it. 🤩 Maybe we can migrate that to [NcAppNavigationCaption](https://nextcloud-vue-components.netlify.app/#/Components/App%20containers/NcAppNavigation?id=ncappnavigationcaption) for the headings and [NcListItem](https://nextcloud-vue-components.netlify.app/#/Components/NcListItems?id=nclistitem) for the announcements? :)我想这是左边栏。我一直梦想着这一点，但从来没有找到时间来实现它。也许我们可以迁移到NcAppNavigationCaption的标题和NcListItem的公告？：）
> 
> Nextcloud Designers also offer Design feedback and app reviews. Every Tuesday 13 UTC. Not sure that is too late for you (guessing your time zone from the language of the screenshots). Nextcloud设计师还提供设计反馈和应用程序评论。每周二13 UTC。不知道这对你来说是否太晚了（从截图的语言中猜测你的时区）。 If you are up for it, I would ask them to give us a spot and then we can walk it through with them at https://cloud.nextcloud.com/index.php/call/gqff69i8如果你准备好了，我会请他们给予我们一个位置，然后我们可以在https://cloud.nextcloud.com/index.php/call/gqff69i8上与他们一起走过。
> 
> There are some technical difficulties with your PR (chmod of all files changed etc) which I guess is because you commited from Windows? It's not a problem, we can revert this together :)有一些技术上的困难与您的公关（chmod的所有文件更改等），我猜这是因为你从Windows提交？这不是一个问题，我们可以一起恢复：）



> Oh wow, that's some fancy and big update!哦，哇，这是一些花式和大更新！ I invited you to our organisation, then you can push the branch directly to the repo and CI will run happily all the time and at the same time we can collaborate on some last fixes. 我邀请您加入我们的组织，然后您可以将分支直接推到存储库，CI将一直愉快地运行，同时我们可以在一些最后的修复上进行协作。
> 
> > 1. add Nextcloud text editor integration添加Nextcloud文本编辑器集成
> 
> That's very fancy, nice!很漂亮，很好！
> 
> > 2. add attachment support 添加附件支持
> > 3. support attachment viewer & download支持附件查看器和下载
> 
> I don't really need that for my usecases, but it's also cool to have!我真的不需要我的用例，但它也很酷！
> 
> > 4. classify announcements by time按时间分类公告
> 
> I guess that's the left sidebar. I was always dreaming of this, but never found the time to implement it. 🤩 Maybe we can migrate that to [NcAppNavigationCaption](https://nextcloud-vue-components.netlify.app/#/Components/App%20containers/NcAppNavigation?id=ncappnavigationcaption) for the headings and [NcListItem](https://nextcloud-vue-components.netlify.app/#/Components/NcListItems?id=nclistitem) for the announcements? :)我想这是左边栏。我一直梦想着这一点，但从来没有找到时间来实现它。也许我们可以迁移到NcAppNavigationCaption的标题和NcListItem的公告？：）
> 
> Nextcloud Designers also offer Design feedback and app reviews. Every Tuesday 13 UTC. Not sure that is too late for you (guessing your time zone from the language of the screenshots). Nextcloud设计师还提供设计反馈和应用程序评论。每周二13 UTC。不知道这对你来说是否太晚了（从截图的语言中猜测你的时区）。 If you are up for it, I would ask them to give us a spot and then we can walk it through with them at https://cloud.nextcloud.com/index.php/call/gqff69i8如果你准备好了，我会请他们给予我们一个位置，然后我们可以在https://cloud.nextcloud.com/index.php/call/gqff69i8上与他们一起走过。
> 
> There are some technical difficulties with your PR (chmod of all files changed etc) which I guess is because you commited from Windows? It's not a problem, we can revert this together :)有一些技术上的困难与您的公关（chmod的所有文件更改等），我猜这是因为你从Windows提交？这不是一个问题，我们可以一起恢复：）

Thank you for your reply and recognition, may I ask how I should merge the code now, I expect there will be some additional updates and maintenance in the future

**Comment by INSIinc** [comment]
I have migrated to NcAppNavigationCaption and NcListItem

**Comment by INSIinc** [comment]
image

**Comment by domdorn** [comment]
This looks nice, hope it gets merged soon!

File: README.md [hunk 1]
```diff
@@ -4,29 +4,19 @@
...
-* My work-focus shifted away from this app
-
-I will continue to provide the level of maintenance I can afford, which is:
-
-* Taking care that the app continues to work
-* Make sure an update is available for compatibility with new Nextcloud server releases
-* Provide feedback and background information to new or existing features
-
-While I'm personally limited in resources to further advance this app I would be more than excited if you want to collaborate with me. I will merge pull requests for new features and frontend fixes.
+Active updates are currently under way
```
Comment (nickvergessen) [inline README.md:22]: We can remove the block completely

File: l10n/es.js
Comment (nickvergessen) [inline l10n/es.js:1]: Please drop these changes. Translation updates are happening automatically via transifex

File: l10n/es.json
Comment (nickvergessen) [inline l10n/es.json:1]: Please drop these changes. Translation updates are happening automatically via transifex

File: l10n/zh_CN.js
Comment (nickvergessen) [inline l10n/zh_CN.js:1]: Please drop these changes. Translation updates are happening automatically via transifex

File: l10n/zh_CN.json
Comment (nickvergessen) [inline l10n/zh_CN.json:1]: Please drop these changes. Translation updates are happening automatically via transifex

File: lib/Controller/AttachmentController.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	 * @NoCSRFRequired
+	 *
+	 */
+	public function getAll($apiVersion)
+	{
+
+		$attachment = $this->attachmentService->findAll($this->request->getParam('announcementId'), true);
+		if ($apiVersion === '1.0') {
+			$attachment = array_filter($attachment, function ($attachment) {
+				return $attachment->getType() === 'deck_file';
```
Comment (nickvergessen) [inline lib/Controller/AttachmentController.php:57]: doesn't look correctly

File: translationtool.phar
Comment (nickvergessen) [inline translationtool.phar:1]: This should not be here

File: translationtool.phar:Zone.Identifier
Comment (nickvergessen) [inline translationtool.phar:Zone.Identifier:1]: This should not be here

File: translationfiles/zh_CN/announcementcenter.po
Comment (nickvergessen) [inline translationfiles/zh_CN/announcementcenter.po:1]: This should not be here

File: translationfiles/templates/announcementcenter.pot
Comment (nickvergessen) [inline translationfiles/templates/announcementcenter.pot:1]: This should not be here

**Review by nickvergessen — COMMENTED** [review]
For me this is too much and I'm having a hard time reviewing this (first because GitHub breaks the browser tab due to the size, second to the content).

It brings in several nice features and a fancy UI, but it has too much code that I do not understand (I'm only a backend developer).

I'd be fine with handing over the app completely to you, if you want to carry on the work, but I don't really want to maintain another share provider and also wouldn't like to see this on the 2 servers I run with this app, due to it's performance impact, but I also don't need comments for my case, so I could fork my "notify about maintenance window" usecase into a slimmed down version of the app without all that:

- Could have the **announcementcenter** with all the features.
- And a **maintenance window announcement** with only "Admin dropping text to all users" (No group selection, no comments, no attachments, no left sidebar, no shares, no editing, etc).

**Comment by nickvergessen** [comment]
I'm closing this as per https://github.com/nextcloud/announcementcenter/pull/725#pullrequestreview-1925033930
I don't see this app adding so much complexity.
It can be merged if someone is taking over the ownership.

**Pull request closed without merging**